### DISCUSSION TITLE
Can WebView support bidirectional communication with its host?

### DISCUSSION THREAD

**Discussion by hismichael** [dotnet/maui#351] [Ideas]
## Summary
As currently the WebView control only provides the Eval() method to evaluate JavaScript code, yet not a mechanism to enable the host to post messages into the Web content, therein they can be handled in the scripting context.

**Comment by HoloShaii** [comment]
I am also currently looking for a way of doing this, the old way of doing it within Xamarin.Forms doesn't seem to work anymore, seeing as it relied on a few classes that are no longer used within .NET MAUI, really hoping there is some workaround, or that it's going to be implemented soon.

**Comment by rbanks54** [comment]
The EvaluateJavaScriptAsync method might be what you're after
https://github.com/dotnet/maui/blob/e2767b257152bd03588e4d11042ee0cbb0c18514/src/Controls/src/Core/WebView.cs#L102

But probably only useful for calling into the webview to return a value.

Ideally, to send messages from the webview back to the host you'd want some kind of event subscription approach. You could possibly approximate the approach by calling EvalJavaScriptAsync() in a loop, but that's not a very usable method.

**Reply by jsuarezruiz** [reply]
And now is already implemented: https://github.com/dotnet/maui/blob/main/src/Core/src/Handlers/WebView/WebViewHandler.cs#L25

**Comment by fairking** [comment]
You can look at the [WebMessageReceived](https://docs.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2.webmessagereceived?view=webview2-dotnet-1.0.1185.39)

I am trying to work it out somehow
https://techcommunity.microsoft.com/t5/app-development/hybrid-spa-application-with-maui-webview/m-p/3287154
